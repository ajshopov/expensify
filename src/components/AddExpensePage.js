import React from "react";

const AddExpensePage = () => (
  <div>
    This is addExpense.
  </div>
);

export default AddExpensePage;