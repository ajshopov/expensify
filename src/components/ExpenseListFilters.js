import React from "react";
import { connect } from "react-redux";
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from "../actions/filters";
import { DateRangePicker } from "react-dates";


class expenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }
  render(){
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={(e) => {
            console.log(e.target.value)
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {
            console.log(e.target.value)
            if (e.target.value === "amount") {
              this.props.dispatch(sortByAmount());
            } else if (e.target.value === "date") {
              this.props.dispatch(sortByDate());
            }
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
        startDate={this.props.filters.startDate}
        startDateId="start"
        endDate={this.props.filters.endDate}
        endDateId="end"
        onDatesChange={this.onDatesChange}
        focusedInput={this.state.calendarFocused}
        onFocusChange={this.onFocusChange}
        numberOfMonths={1}
        isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    );
  }
};

//  what do we need from the store? filters
const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(expenseListFilters);