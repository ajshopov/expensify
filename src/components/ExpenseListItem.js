import React from "react";
import { Link } from "react-router-dom";

//  destructuring format
// const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
const ExpenseListItem = (props) => (
  <div>
    <h4><Link to={`/edit/${props.id}`}>{props.description}</Link></h4>
    <p>{props.amount} - {props.createdAt}</p>
  </div>
);

export default ExpenseListItem;