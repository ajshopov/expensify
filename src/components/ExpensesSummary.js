import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const pluralise = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedAmount = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div className="summary">
      <div className="content-container">
        <h2 className="summary__title">
          Viewing <span>{expenseCount}</span> {pluralise} totalling <span>{formattedAmount}</span>
        </h2>
        <Link className="button" to="/create">Add Expense</Link>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  // console.log(state);
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);