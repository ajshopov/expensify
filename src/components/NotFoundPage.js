import React from "react";
import { Link } from "react-router-dom";

const NotFoundPage = () => (
  <div>
    This is NotFoundPage.
    <Link to="/">Go Home</Link>
  </div>
);

export default NotFoundPage;