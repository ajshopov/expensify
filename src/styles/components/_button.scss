.button{
    background: $blue;
    border: none;
    color: white;
    padding: $s-size;
    font-weight: 300;
    font-size: $font-size-large;
    display: inline-block;
    font-weight: 300;
    text-decoration: none;
    margin-top: $m-size;
    line-height: 1;
}

.button--secondary{
    background: #888;
}