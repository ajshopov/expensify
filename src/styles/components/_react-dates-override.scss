
.DateRangePicker, .SingleDatePicker{
    display: block;
}
.DateRangePickerInput, .SingleDatePickerInput{
    width: 100%;
}
.SingleDatePickerInput > .DateInput{
    width: 100%;
}
