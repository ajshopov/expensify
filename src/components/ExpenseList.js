import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";

//  regular unconnected component
export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
        props.expenses.map((expense) => {
          return <ExpenseListItem key={expense.id} {...expense} />
        })
      )
    }
  </div>
);

// mapping state to props
const mapStateToProps = (state) => {
  console.log(state);
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

//  call to connect redux
export default connect(mapStateToProps)(ExpenseList);
