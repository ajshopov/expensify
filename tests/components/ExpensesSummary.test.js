import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../src/components/ExpensesSummary";

test('should correctly render expenses summary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={230} />);
  expect(wrapper).toMatchSnapshot();
})

test('should correctly render expenses summary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={10} expensesTotal={114200} />);
  expect(wrapper).toMatchSnapshot();
})
