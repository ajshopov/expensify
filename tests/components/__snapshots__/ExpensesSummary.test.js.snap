// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    10
     
    expenses
     totalling 
    $1,142.00
  </p>
</div>
`;

exports[`should correctly render expenses summary with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $2.30
  </p>
</div>
`;
